File: server/core/src/main/java/dev/slimevr/websocketapi/WebsocketConnection.java
package dev.slimevr.websocketapi;

import dev.slimevr.protocol.ConnectionContext;
import dev.slimevr.protocol.GenericConnection;
import org.java_websocket.WebSocket;
import org.java_websocket.exceptions.WebsocketNotConnectedException;

import java.nio.ByteBuffer;
import java.util.UUID;


public class WebsocketConnection implements GenericConnection {

	public final ConnectionContext context;
	public final WebSocket conn;
	public UUID id;

	public WebsocketConnection(WebSocket conn) {
		this.context = new ConnectionContext();
		this.conn = conn;
		this.id = UUID.randomUUID();
	}

	@Override
	public ConnectionContext getContext() {
		return this.context;
	}

	@Override
	public void send(ByteBuffer bytes) {
		if (this.conn.isOpen()) {
			try {
				this.conn.send(bytes.slice());
			} catch (WebsocketNotConnectedException ignored) {
				// Race condition if it closes between our check and sending
			}
		}
	}

	@Override
	public UUID getConnectionId() {
		return id;
	}
}

File: server/core/src/main/java/dev/slimevr/websocketapi/WebsocketConnection.kt
package dev.slimevr.websocketapi

import dev.slimevr.protocol.ConnectionContext
import dev.slimevr.protocol.GenericConnection
import org.java_websocket.WebSocket
import org.java_websocket.exceptions.WebsocketNotConnectedException
import java.nio.ByteBuffer
import java.util.*

class WebsocketConnection(val conn: WebSocket) : GenericConnection {
	override val context = ConnectionContext()
	override val connectionId: UUID = UUID.randomUUID()

	override fun send(bytes: ByteBuffer) {
		if (this.conn.isOpen) {
			try {
				this.conn.send(bytes.slice())
			} catch (ignored: WebsocketNotConnectedException) {
				// Race condition if it closes between our check and sending
			}
		}
	}
}
